### DISCUSSION TITLE
Return trace id from Jaeger in response header

### DISCUSSION THREAD

**Discussion by omerbarlev** [jaegertracing/jaeger#3783] [Q&A] [answered]
I am using opentelemetry in my python FastAPI application. It sends traces back to Jaeger. Each trace has a unique trace id I can also search for in the Jaeger UI search bar.

I want to return in the response headers that trace id so that later on I can look a trace for a specific request (just take the trace id from the header and search for it in Jaeger UI).

I tried adding a middleware in FastAPI that adds the trace id like that -

    class EnrichOpenTelemetryId(BaseHTTPMiddleware):
        def __init__(self, app):
            super().__init__(app)
    
        async def dispatch(self, request: Request, call_next):
            with tracer.start_span("dummy-span") as span:
                trace_id = str(span.get_span_context().trace_id)
    
            response = await call_next(request)
            response.headers["X-Trace-Id"] = trace_id
    
            return response   

There is a new header containing some trace id (e.g - 149906749482483391829634904508866243046) in each request but it's not the one Jaeger uses... (meaning I cannot search on that trace id). Jaeger trace id looks something like that - 8eddc793de380fa76c54557af09538e3.

So how do I get Jaeger trace id?

**Comment by yurishkuro** [comment, marked as answer]
You're getting an integer back, you need to convert it to hex value.

**Reply by omerbarlev** [reply]
Hi, thank you.
It seems to be working but now I see there are a lot of new traces created in Jaeger.
I tried adding the trace id like this - 

    async def dispatch(self, request: Request, call_next):
        with tracer.start_span("dummy-span") as span:
            trace_id = span.get_span_context().trace_id

        response = await call_next(request)
        response.headers["X-Trace-Id"] = format(trace_id, 'x')

        return response
        
And I do get a the right trace id in the response headers.
But now, there are many "dummy-span" traces created containing almost no data.
I would like those traces to not be created at all... 

Here is an example of unwanted created span - 
![image](https://user-images.githubusercontent.com/93929968/176303963-159968e0-967a-4e1d-af82-911b136a5a55.png)

**Reply by yurishkuro** [reply]
When you initialize the tracer, you give it a context manager, which has a method for retrieving the current span. So you don't need to create a new span just to obtain a trace ID.

**Reply by omerbarlev** [reply]
amazing, thank you so much!
